.github/workflows/update-labels.yml

```
@@ -28,3 +30,33 @@ jobs:
               repo: context.repo.repo,
               name: "wait for update"
             })
+
+  pr_commented:
+    if: github.event.issue && github.event.issue.pull_request && github.event.comment.user.login == github.event.issue.user.login && (contains(github.event.issue.labels.*.name, 'wait for update') || contains(github.event.issue.labels.*.name, 'discuss')) && !contains(github.event.issue.labels.*.name, 'user responded')
+    runs-on: ubuntu-latest
+    steps:
+      - name: update label when user responds
+        uses: actions/github-script@v6
+        with:
+          script: |
+            github.rest.issues.addLabels({
+              issue_number: context.issue.number,
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              labels: ["user responded"]
+            })
+
+  pr_review_commented:
+    if: github.event.pull_request && github.event.comment.user.login == github.event.pull_request.user.login && (contains(github.event.pull_request.labels.*.name, 'wait for update') || contains(github.event.pull_request.labels.*.name, 'discuss')) && !contains(github.event.pull_request.labels.*.name, 'user responded')
```

I think it should be:
```suggestion
    if: github.event.pull_request && github.event.pull_request_review_comment.user.login == github.event.pull_request.user.login && (contains(github.event.pull_request.labels.*.name, 'wait for update') || contains(github.event.pull_request.labels.*.name, 'discuss')) && !contains(github.event.pull_request.labels.*.name, 'user responded')
```

https://docs.github.com/en/webhooks/webhook-events-and-payloads#pull_request_review_comment

---

The `github.event` stands for the event `pull_request_review_comment`, so we just need to get the property `comment` from `github.event`

---

I hope you performed a manual test/demo in a dummy repository. It's a good idea to do so.

---

Yeah, i metioned this in the description
```
I tested in the pr: https://github.com/monkeyDluffy6017/apisix/pull/6
```